qt/widgets/common/src/SelectFunctionDialog.cpp

```diff
@@ -89,14 +89,62 @@ void SelectFunctionDialog::constructFunctionTree(
     }
   };
 
+  // keeps track of categories added to the tree
+  QMap<QString, QTreeWidgetItem *> categories;
+
   for (const auto &entry : categoryFunctionsMap) {
-    if (showCategory(entry.first)) {
-      QTreeWidgetItem *category = new QTreeWidgetItem(m_form->fitTree);
-      category->setText(0, QString::fromStdString(entry.first));
 
-      for (const auto &function : entry.second) {
-        QTreeWidgetItem *fit = new QTreeWidgetItem(category);
-        fit->setText(0, QString::fromStdString(function));
+    QString categoryName = QString::fromStdString(entry.first);
+    QStringList subCategories = categoryName.split('\\');
+    if (!categories.contains(categoryName)) {
+      if (subCategories.size() == 1) {
+        if (showCategory(entry.first)) {
+          QTreeWidgetItem *catItem =
+              new QTreeWidgetItem(QStringList(categoryName));
+          categories.insert(categoryName, catItem);
+          m_form->fitTree->addTopLevelItem(catItem);
+          for (const auto &function : entry.second) {
+            QTreeWidgetItem *fit = new QTreeWidgetItem(catItem);
+            fit->setText(0, QString::fromStdString(function));
+          }
+        }
+      } else {
+        // go through the path and add the folders if they don't already exist
+        QString currentPath = subCategories[0];
+        QTreeWidgetItem *catItem = nullptr;
+        int n = subCategories.size();
```

Comment:
Would help for readability if something more descriptive was used in place of `int n`, especially as its used further down in the block.